doc/protocols/scion-header.rst

```diff
@@ -787,19 +787,33 @@ Colibri Packet Timestamp
      0                   1                   2                   3
      0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    |                             TsRel                             |
+    |                        Expiration Tick                        |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                TsRel / Original Payload Length                |
```

Comment:
Regarding the size of the timestamp (TsRel and PckId):
I think in principle it could be made a few bytes smaller.
However, when doing this we should keep in mind that
- In the future there might be high-performance implementations that send significantly faster than our Go-based implementation.
- The PckId is still necessary to avoid collisions among different cores or applications.
- The start of the info field might then not be 4-byte aligned anymore.
- For consistency it might make sense to keep the same size as the EPIC timestamp (8 bytes), so that the duplicate suppression system does not need to distinguish between different timestamps of different path types.

Note that the Expiration Tick cannot be made smaller, as it already requires today 4 bytes just to represent the current time.

Reply:
Update:
Should we decide to make the timestamp shorter, here a few ideas:
- TsRel only needs to encode the time the packet was sent with a granularity that allows to check that the packet is current. With 2 bytes, a precision of 0,25 ms can be achieved (1B would be too imprecise).
- When reducing the size of TsRel, the PckId becomes more important. I think 4 bytes for TsRel should be realistic such that an AS can have multiple gateways with multiple cores, where each core keeps its own packet counter. The AS is free to dedicate X bytes to identify the gateway/core/application/... and 4-X bytes to identify the packet.